Assets/Scripts/Managers/PauseManager.cs

```diff
@@ -0,0 +1,110 @@
+// Morgan Houston
+using StarterAssets;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+
+public class PauseManager : MonoBehaviour
+{
+    #region Fields
+
+    public GameObject pauseMenu;
+    public Button resumeButton, mainMenuButton;
+
+    private StarterAssetsInputs inputs;
+    private PlayerInput playerInput;
+    [SerializeField]
+    private bool isGamePaused;
+
+    #endregion
+
+    #region Unity Methods
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        inputs = GetComponent<StarterAssetsInputs>();
+        playerInput = GetComponent<PlayerInput>();
+        isGamePaused = false;
+        Cursor.lockState = CursorLockMode.Locked;
+    }
+
+    private void OnEnable()
+    {
+        resumeButton?.onClick.AddListener(ResumeGame);
+        mainMenuButton?.onClick.AddListener(MainMenu);
+    }
+
+    private void OnDisable()
+    {
+        resumeButton?.onClick.RemoveListener(ResumeGame);
+        mainMenuButton?.onClick.RemoveListener(MainMenu);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        CheckIfPause();
+    }
+
+    #endregion
+
+    #region Methods
+
+    private void CheckIfPause()
+    {
+        if (inputs != null && inputs.pause)
+        {
+            isGamePaused = !isGamePaused;
+            inputs.pause = false;
+        }
+
+        SetPauseMenu();
+
+    }
+
+    private void SetPauseMenu()
+    {
+        // PAUSED
+        if (isGamePaused && !pauseMenu.activeInHierarchy)
+        {
+            Pause();
+        }
+        // NOT PAUSED
+        else if (!isGamePaused && pauseMenu.activeInHierarchy)
+        {
+            Resume();
+        }
+    }
+
+    private void Pause()
+    {
```

Comment:
Pause is not pausing physics, the character stills fall down when paused.
¿Should we add Time.timesale=0 when pausing and 1 to resume?